lib/blueprinter/base.rb

```diff
@@ -42,7 +43,8 @@ def self.prepare(object, view:)
       unless view_collection.has_view? view_name
         raise BlueprinterError, "View '#{view_name}' is not defined"
       end
-      prepared_object = select_columns(object, view_name: view_name)
+      prepared_object = Optimizer.new(object)
+                          .select(view_collection.fields_for(view))
```

Comment:
I think I would prefer to abstract this method to something like `optimize!` that then calls select, since we don't know at this point that this is an object that we should call `select` on. Plus, it allows us to expand the `Optimizer` later if we add more ways to optimize than just field selection

Reply:
makes perfect sense!